modules/internal/ChapelDistribution.chpl

```diff
@@ -607,8 +629,13 @@ module ChapelDistribution {
   }
 
   // Currently this is not optimized for addition of a sparse
+  // This assumes that a domain is sorted by row, which may not always be true.
+  // There could be domains sorted by column instead.
+  // It would be nice if there was a common interface in BaseDom to query
+  // whether a domain was row-major or column-major to distinguish these.
+  // The current implementation is a work-around for this shortcoming.
   proc +=(ref sd: domain, d: domain)
-  where isSparseDom(sd) && d.rank==sd.rank && sd.idxType==d.idxType {
+  where sd._value: DefaultSparseDom && d.rank==sd.rank && sd.idxType==d.idxType {
```

Comment:
Perhaps my comment was not clear enough. We *do* support `+=` on `CSDom`, but it is an awkward work-around (implemented at the bottom of `LayoutCS.chpl`)